src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -0,0 +1,173 @@
+    /// NOTE: If COMPlus_EnableEventPipe != 4 then this listener is not created and does not add any overhead to the process.
+    /// </summary>
+    internal sealed class EventPipeController
+    {
+        private const string MarkerFileExtension = ".ctl";
+        private const int PollingIntervalMilliseconds = 10000; // 10 seconds
+        private const uint DefaultCircularBufferMB = 1024; // 1 GB
+        private readonly char[] ProviderConfigDelimiter = new char[] { ',' };
+        private readonly char[] ConfigComponentDelimiter = new char[] { ':' };
+
+        private readonly EventPipeProviderConfiguration[] DefaultProviderConfiguration = new EventPipeProviderConfiguration[]
+        {
+            new EventPipeProviderConfiguration("Microsoft-Windows-DotNETRuntime", 0x4c14fccbd, 5),
+            new EventPipeProviderConfiguration("Microsoft-Windows-DotNETRuntimePrivate", 0x4002000b, 5),
+            new EventPipeProviderConfiguration("Microsoft-DotNETCore-SampleProfiler", 0x0, 5)
+        };
+
+        private static EventPipeController s_controllerInstance;
+
+        private string m_traceFilePath;
+        private string m_markerFilePath;
+        private bool m_markerFileExists;
+
+        internal static void Initialize()
+        {
+            // Don't allow failures to propagate upstream.
+            // Instead, ensure program correctness without tracing.
+            try
+            {
+                if (s_controllerInstance == null)
+                {
+                    string strEnabledValue = CompatibilitySwitch.GetValueInternal("EnableEventPipe");
+                    if (strEnabledValue != null)
+                    {
+                        int enabledValue = Convert.ToInt32(strEnabledValue);
+                        if (enabledValue == 4)
+                        {
+                            s_controllerInstance = new EventPipeController();
+                        }
+                    }
+                }
+            }
+            catch { }
+        }
+
+        private EventPipeController()
+        {
+            // Determine the marker file path.
+            string traceFilePath = CompatibilitySwitch.GetValueInternal("EventPipeOutputFile");
+            if (!string.IsNullOrEmpty(traceFilePath))
+            {
+                m_traceFilePath = traceFilePath;
+                m_markerFilePath = traceFilePath + MarkerFileExtension;
+            }
+            else
+            {
+                // If the output file path is not specified then use
+                // path/to/current/working/directory/Process-<pid>.netperf.ctl
+                m_traceFilePath = EventPipeInternal.GetDefaultTraceFileName();
+                m_markerFilePath = m_traceFilePath + MarkerFileExtension;
+            }
+
+            m_markerFileExists = false;
+
+            // Start a new thread to listen for tracing commands.
+            Task.Factory.StartNew(WaitForTracingCommand, TaskCreationOptions.LongRunning);
```

Comment:
We can avoid burning the thread by 

     m_timer = new Timer(WaitForTracingCommand, null, 10000, 10000);

Which achieves the same effect but does not uses threadpool threads (and only for very short durations).